graphql_compiler/macros/macro_expansion.py

```diff
@@ -323,18 +330,15 @@ class name -> (macro edge name -> MacroEdgeDescriptor)
                     macro_registry, inherited_macro_edges, vertex_field_type,
                     new_selection_ast, new_query_args, tag_names)
 
-        if new_selection_ast is not selection_ast:
+        if new_selection_ast is not selection_ast or selections_before or selections_after:
```

Comment:
Could we have a situation where `new_selection_ast is not selection_ast` is False but the rest of this is `True`? Am I missing something, or should this maybe be an `AssertionError`? If there is such a case, it's probably a good idea to both describe it explicitly in a comment here, and also include a test case for it, since it's far from obvious.